src/ResultsPage.js

```diff
@@ -47,4 +46,6 @@ const ResultsPage = () => {
     );
 };
 
-export default ResultsPage;
\ No newline at end of file
+export default connect(({subjects}) => ({
+    subjects
+}))(ResultsPage);
```

Comment:
Ta dekonstrukce parametru `state` na `({subjects})` je jako hezká JS feature, ale popravdě mě to v tý změti závorek trochu zmátlo :D Tady je to sice triviální, ale jako příklad pro nás ostatní (abychom to mohli replikovat jinde) by to bylo možná pochopitelnější takhle
```
connect(state => {
    subjects: state.subjects
})(ResultsPage)
```

Pak je vidět, že ta funkce předaná do `connect()` teda jednoduše žere redux state, ve kterém jsou subjects, a vrací props jako slovník.

Reply:
Jasný, on to může být ještě složitější. Pravda je, že jako example začnu od nejobecnějšího a budu ubírat... V kódu to nechám jak to bude finálně (tj. takto). Naprosto chápu, že tě to zmátlo. JS má hodně skratek, které dělají věci implicitně.